lib/ff_dpdk_if.c

```diff
@@ -424,13 +432,13 @@ init_arp_ring(void)
     int proc_id = ff_global_cfg.dpdk.proc_id;
 
     /* Allocate arp ring ptr according to eth dev count. */
-    int nb_ports = rte_eth_dev_count();
+    int nb_dev_ports = rte_eth_dev_count();
```

Comment:
It's legacy here, you can just use ` int nb_ports = ff_global_cfg.dpdk.nb_ports;`.

Reply:
here we need use `max_portid`

Reply:
Why? `max_portid+1` is not necessarily equal to `nb_ports`. This place needs the count of ports.

Reply:
because here we actually need the size of `struct rte_ring*` array and this array's index is `port_id`, so we must use `max_portid+1`, I think `nb_ports` is not a good name here, maybe `ring_arr_sz` is better.